### PR TITLE
(@wdio/utils): Add envDetector for browserstack

### PR DESCRIPTION
**Pull request by Pi-fe**
## Proposed changes

[//]: # I fix the bug that doesn't allow to check isMobile, isAndroid, isIOS command in browserstack

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-utils/src/envDetector.ts [hunk 1]
```diff
@@ -107,10 +111,12 @@ function isMobile(capabilities: WebdriverIO.Capabilities) {
          * browserName is empty (and eventually app is defined)
          */
         capabilities.browserName === '' ||
+        capabilities['bstack:options']?.browserName === '' ||
```
Comment (christian-bromann) [inline packages/wdio-utils/src/envDetector.ts:114]:
```suggestion
        bsOptions.browserName === '' ||
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Review by christian-bromann — COMMENTED** [review]
If `capabilities['bstack:options'].platformName` is a thing we should extend the BrowserStackCapabilities type https://github.com/webdriverio/webdriverio/blob/2b6cea304eea672908ade21ae49b7919da123df5/packages/wdio-types/src/Capabilities.ts#L1095.

**Comment by Pi-fe** [comment]
> If `capabilities['bstack:options'].platformName` is a thing we should extend the BrowserStackCapabilities type
> 
> https://github.com/webdriverio/webdriverio/blob/2b6cea304eea672908ade21ae49b7919da123df5/packages/wdio-types/src/Capabilities.ts#L1095
> 
> .

Ok, I add this to BSCapabilities

File: packages/wdio-utils/src/envDetector.ts [hunk 2]
```diff
@@ -99,6 +100,9 @@ function isMobile(capabilities: WebdriverIO.Capabilities) {
         capabilities.platformName && capabilities.platformName.match(/ios/i) ||
         capabilities.platformName && capabilities.platformName.match(/tvos/i) ||
         capabilities.platformName && capabilities.platformName.match(/android/i) ||
+        /ios/i.test(bsOptions.platformName) ||
```
Comment (christian-bromann) [inline packages/wdio-utils/src/envDetector.ts:103]:
```suggestion
        /ios/i.test(bsOptions.platformName || '') ||
```

**Comment by Pi-fe** [comment]
Ok I fixed all ""blank"" properties

File: packages/wdio-utils/src/envDetector.ts [hunk 3]
```diff
@@ -120,13 +126,16 @@ function isMobile(capabilities: WebdriverIO.Capabilities) {
  * @return {Boolean}               true if run on iOS device
  */
 function isIOS(capabilities?: Capabilities.DesiredCapabilities) {
+    const bsOptions = capabilities['bstack:options'] || {}
```
Comment (christian-bromann) [inline packages/wdio-utils/src/envDetector.ts:129]:
```suggestion
    const bsOptions = capabilities?.['bstack:options'] || {}
```

File: packages/wdio-utils/src/envDetector.ts [hunk 4]
```diff
@@ -136,12 +145,15 @@ function isIOS(capabilities?: Capabilities.DesiredCapabilities) {
  * @return {Boolean}               true if run on Android device
  */
 function isAndroid(capabilities?: WebdriverIO.Capabilities) {
+    const bsOptions = capabilities['bstack:options'] || {}
```
Comment (christian-bromann) [inline packages/wdio-utils/src/envDetector.ts:148]:
```suggestion
    const bsOptions = capabilities?.['bstack:options'] || {}
```

**Review by christian-bromann — COMMENTED** [review]
fixing types

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
Congratulations on your remarkable first contribution to WebdriverIO! This project thrives on the invaluable involvement of our community, and we are truly grateful for your contribution. We eagerly anticipate witnessing more of your exceptional work, so please don't hesitate to inform us if we can assist you in identifying intriguing areas where you can make further contributions. Join our lively [Discord](https://discord.webdriver.io/) channel and reach out to us; we would be delighted to connect with you. Your efforts are deeply appreciated, and we extend our heartfelt gratitude to you. 🙏 ❤️

**Comment by christian-bromann** [comment]
fixes #11474

**Comment by Pi-fe** [comment]
> Congratulazioni per il tuo straordinario primo contributo a WebdriverIO! Questo progetto si basa sul prezioso coinvolgimento della nostra comunità e siamo davvero grati per il tuo contributo. Non vediamo l'ora di assistere ad altro del tuo eccezionale lavoro, quindi non esitare a informarci se possiamo aiutarti a identificare aree interessanti in cui puoi dare ulteriori contributi. Unisciti al nostro vivace canale [Discord](https://discord.webdriver.io/) e contattaci; saremo lieti di entrare in contatto con te. I vostri sforzi sono profondamente apprezzati e vi esprimiamo la nostra più sincera gratitudine. 🙏 ❤️

Thank you very much, as a mobile tester automater, I choose wdio + appium for my project!
Every time I'll have a suggestion or problem I'll write on github or discord or I try to fix.

Good job! :P

**Pull request merged**